pkg/spoke/statusfeedback/rules/rule.go

```diff
@@ -0,0 +1,41 @@
+package rules
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	workapiv1 "open-cluster-management.io/api/work/v1"
+)
+
+type WellKnownStatusRuleResolver interface {
+	GetPathsByKind(schema.GroupKind) []workapiv1.JsonPath
+}
+
+type DefaultWellKnownStatusResolver struct {
+	rules map[schema.GroupKind][]workapiv1.JsonPath
+}
+
+var deploymentRule = []workapiv1.JsonPath{
+	{
+		Name: "ReadyReplicas",
+		Path: ".status.readyReplicas",
```

Comment:
this path is logically dependent on the version.  The request to get status from an object in the API is logically *not* dependent on version (groupKind), but by the time you get the point of selecting what jsonpath to use, you have the group*version*kind and need to use that.

Reply:
right, changed to use GVK as the key here